javaCarAPI/src/main/java/api/CarAPI.java

```diff
@@ -0,0 +1,261 @@
+    private String jsonCommand;
+    private String ultrasonicFront;
+    private String gyroscope;
+    private String infrared;
+    private String odometerSpeed;
+    private String odometerTotalDistance;
+    private String heartbeat;
+
+    private final MqttClient client;
+    private final String carName;
+    private final String clientId;
+
+    private long ping = 0;
+
+    private final List<Consumer<Image>> videoFeedListeners = new ArrayList<>();
+    private final List<Consumer<Double>> ultraSonicListeners = new ArrayList<>();
+    private final Map<Infrared,List<Consumer<Double>>> infraredListeners = new HashMap<>();
+    private final Map<Odometer,List<Consumer<Double>>> odometerSpeedListeners = new HashMap<>();
+    private final Map<Odometer,List<Consumer<Double>>> odometerTotalDistanceListeners = new HashMap<>();
+    private final List<Consumer<Double>> gyroscopeListeners = new ArrayList<>();
+
+    public CarAPI(String host, String carName) throws MqttException{
+        this(host, UUID.randomUUID().toString(),carName);
+    }
+
+    public CarAPI (String host,String clientId,String carName) throws MqttException{
+        this(host,1883,clientId,carName);
+    }
+
+    public CarAPI (String host,int port,String clientId,String carName) throws MqttException{
+        this(host,port,clientId,carName,null,null);
+    }
+
+    public CarAPI (String host,int port,String clientId,String carName,String user, char [] password) throws MqttException{
+        this.carName = carName;
+        this.clientId = clientId;
+
+        initMQTTVariables();
+
+        String hostURL = "tcp://"+host+":"+port;
+        client = new MqttClient(hostURL,clientId, new MqttDefaultFilePersistence(System.getProperty("java.io.tmpdir")));
+        MqttConnectOptions options = new MqttConnectOptions();
+        options.setAutomaticReconnect(true);
+        options.setCleanSession(true);
+        options.setConnectionTimeout(10);
+        if (user != null && password != null) {
+            options.setUserName(user);
+            options.setPassword(password);
+        }
+        client.connect(options);
+
+        subscribe();
+        startHeartbeatThread();
+
+    }
+
+    private void startHeartbeatThread() {
+        Thread thread = new Thread(() -> {
+            while (client.isConnected()) {
+                try {
+                    MqttMessage currentTime = new MqttMessage(Long.toString(System.currentTimeMillis()).getBytes(StandardCharsets.UTF_8));
+                    client.publish(heartbeat,currentTime);
+                    final int PING_REQUEST_INTERVAL = 150;
+                    Thread.sleep(PING_REQUEST_INTERVAL);
+                } catch (Exception ignore) {}
+            }
+        });
+        thread.start();
+    }
+
+    private void subscribe() throws MqttException{
+
+
+        client.subscribe(ultrasonicFront,(t,m) -> ultraSonicListeners.forEach(callback -> callback.accept(convertPayloadToDouble(m))));
```

Comment:
I would strongly encourage renaming lambda variables from single letters to something more meaningful just to make the code more readable/easy to debug.